src/app-layer-htp-file.c

```diff
@@ -113,6 +113,12 @@ int HTPFileOpen(HtpState *s, const uint8_t *filename, uint16_t filename_len,
         }
 
         sbcfg = &s->cfg->response.sbcfg;
+        if (s->file_range_ids != 0) {
+            //TODO even better : handle intersected downloads
```

Comment:
what would it take to support this? Files use a unique identifier I think, was it the etag?

Reply:
There are two types of intersected downloads :
- range download intersected with regular download
- range downloads intersected with themselves

For the first case, it only takes a boolean to know if we must call `FileAppendDataById` for the reassembled file
https://github.com/OISF/suricata/pull/4562/files#diff-d5d8efe82303f94955b7b0c979f6585aR364

For the second case, we will a dynamic array for `file_range_ids`